ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto

```diff
@@ -251,4 +251,16 @@ message DisclosedContract {
   // Maintainers of the contract key, if defined.
   // Optional
   repeated string key_maintainers = 9;
+
+  // The contract arguments as a typed record. The fields comprising the create arguments are cross-validated
+  // with the contents of the ``contract_create_arguments_blob``. If any inconsistency is determined,
+  // the command submission is rejected. It can be left unset.
+  // Optional
```

Comment:
@mziolekda : note that the same kind of checking is required for the `signatories`, `observers`, `contract_key`, and `key_maintainers`. Will that be done too? Consider moving the comment about the checks to the `contract_create_arguments_blob`.

Also consider going the alternative way suggested in https://docs.google.com/document/d/1uhDEAKLOqVHXiNOUNcNtABlAzvo1ELRdnDEXOWVIQyg/edit?disco=AAAA-QEyClA where only the minimal amount of data is passed in for a disclosed contract, and decoding is done via an explicit rpc offered by the Ledger API server.

This minimizes encoding problems. One that I can see lurking is actually the `contract_key` of type `Value`, which is likely as hard to pass via clients w/o loss of information as the `create_arguments`.

Reply:
Actually, this approach would also have the benefit of not requiring any change to the API. Just change the commentary that only the `create_arguments_blob` will be supported with upgrading, and mark the `create_arguments` with `[deprecated = true]`.